usaspending_api/references/v2/views/total_budgetary_resources.py

```diff
@@ -0,0 +1,44 @@
+from django.db.models import Sum
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.references.models.gtas_sf133_balances import GTASSF133Balances
+
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from usaspending_api.common.exceptions import InvalidParameterException, UnprocessableEntityException
+
+
+class TotalBudgetaryResources(APIView):
+    """
+    This route sends a request to the backend to retrieve GTAS totals by FY/FP.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/references/total_budgetary_resources.md"
+
+    @cache_response()
+    def get(self, request):
+        fiscal_year = request.query_params.get("fiscal_year")
+        fiscal_period = request.query_params.get("fiscal_period")
+        gtas_queryset = GTASSF133Balances.objects.values("fiscal_year", "fiscal_period")
+
+        if fiscal_period:
+            if not fiscal_year:
+                raise InvalidParameterException("fiscal_period was provided without any fiscal_year.")
+            else:
+                if int(fiscal_period) < 1 or int(fiscal_period) > 12:
+                    raise UnprocessableEntityException("fiscal_period must be in the range 1-12")
+                gtas_queryset = gtas_queryset.filter(fiscal_year=fiscal_year, fiscal_period=fiscal_period)
+
+        elif fiscal_year:
+            gtas_queryset = gtas_queryset.filter(fiscal_year=fiscal_year)
+
+        results = []
+        for gtas in gtas_queryset.annotate(total_budgetary_resources=Sum("total_budgetary_resources_cpe")):
+            results.append(
+                {
+                    "fiscal_year": gtas["fiscal_year"],
+                    "fiscal_period": gtas["fiscal_period"],
+                    "total_budgetary_resources": float(gtas["total_budgetary_resources"]),

```

Comment:
I don't think we want to type this as a float. There is a potential for introducing rounding errors

Reply:
Makes sense, I just copied from toptier_agencies. What is preferred?

Reply:
Interesting. A quick search only revealed two endpoints which utilize python's `float`. Did you test the behavior without any explicit type conversion in the view? This is the very old PR introducing the `float()` https://github.com/fedspendingtransparency/usaspending-api/pull/445 and it seems based on the comments that the author was confused about what is returned by the API (`Decimal` is a python datatype, so it would be returned in tests but not by the api service)